src/mosaic-node-manager.cc

```diff
@@ -99,9 +98,36 @@ namespace ns3 {
             BuildingsHelper::Install (eNodeB);
             BuildingsHelper::MakeMobilityModelConsistent();  
 
-            m_groupL2Address = 0x00;
+            m_groupL2Address = 0x01;
             Ipv4AddressGenerator::Init(Ipv4Address ("10.1.0.0"), Ipv4Mask("255.255.0.0"));
             m_clientRespondersAddress = Ipv4AddressGenerator::NextAddress (Ipv4Mask ("255.255.0.0"));
+
+            // Sidelink configuration
+            m_ueSidelinkConfiguration = CreateObject<LteUeRrcSl>();
+            m_ueSidelinkConfiguration->SetSlEnabled(true);
+            m_ueSidelinkConfiguration->SetV2xEnabled(true);
+
+            LteRrcSap::SlV2xPreconfiguration preconfiguration;
+            preconfiguration.v2xPreconfigFreqList.freq[0].v2xCommPreconfigGeneral.carrierFreq = 54890;
+            preconfiguration.v2xPreconfigFreqList.freq[0].v2xCommPreconfigGeneral.slBandwidth = 30;
+            
+            preconfiguration.v2xPreconfigFreqList.freq[0].v2xCommTxPoolList.nbPools = 1;
+            preconfiguration.v2xPreconfigFreqList.freq[0].v2xCommRxPoolList.nbPools = 1;
+
+            SlV2xPreconfigPoolFactory pFactory;
+            pFactory.SetHaveUeSelectedResourceConfig (true);
```

Comment:
Just a thought, ideally these would be configurable. If that's a trivial change, let's do it now, otherwise let's make a JIRA story to track that it still needs to be done later.

Reply:
The necessary changes might extend beyond just the NS3-Federate; they could also affect the MOSAIC NS3-Ambassador. This is because the current configuration messages sent from the NS3-Ambassador to the NS3-Federate don't encompass those attributes. Would it be possible to make another story to track?

Reply:
Yeah, lets' just track it that way then